cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -839,14 +854,16 @@ public void workflowSchedules(HttpRequest request, HttpResponder responder,
    * Get schedule state.
    */
   @GET
-  @Path("/apps/{app-id}/workflows/{workflow-id}/schedules/{schedule-id}/status")
+  @Path("/apps/{app-id}/workflows/{workflow-id}/schedules/{schedule-name}/status")
```

Comment:
But because this is such a large PR, and other PRs are dependent on it, it is better to address that in a new PR.